db/seeds.rb

```diff
@@ -79,15 +79,54 @@ def position(type)
   surname = player['web_name']
   full_name = "#{player['first_name']} #{player['second_name']}"
   position = position(player['element_type'])
+
+   # Get specific player url
+   url = 'https://fantasy.premierleague.com/drf/element-summary/' << player['id'].to_s
+   response = HTTParty.get(url)
+ 
+    # Get specific player url
+    url = 'https://fantasy.premierleague.com/drf/element-summary/' << player['id'].to_s
+    response = HTTParty.get(url)
+  
+    # Gives hisory of the player
+    history_raw = JSON.parse(response.body)['history']
+ 
+  
+    # Gives the round number as the key
+    round_as_key = history_raw.collect { |p| {p['round'] => p['total_points'] }}
+  
+    # Consolidates keys, adds round number when there are multiple keys
+    points_tidy = round_as_key.each_with_object({}) { |h,g| g.update(h) { |_,o,n| o+n } }
```

Comment:
this whole section, i'm not sure i'm convinced that you know exactly what this is doing

it seems very complicated for something that coud be quite simple.

try this: `history.map { |gw| ["points_in_gameweek_#{gw['round']}", gw['total_points']] }.to_h`

this is what you were doing before, but in one much more readable line. so we map over our history array, creating an array of arrays, where each array looks like this `['points_in_gameweek_1', 1]`, etc. (see how i'm using string interpolation again here). `to_h` is then used to create a hash from our array. Ruby can be beautiful like this

Reply:
Line 92 adds together when players have played more than 1 game in the same gameweek.

ie. Petr Cech would have {... , 37 => 6, 37 => 3}. This then makes it {...., 37 => 9}. 

I'll update line 95 because I think that's what the comment is actually talking about. No I mainly don't know what I'm doing, just knowledge I gained from stackexchange! I've updated that.